### PR TITLE
Fwsp1 103 image test

### PR DESCRIPTION
**Pull request by xendk**

### CODE REVIEW

File: features/README.md [hunk 1]
```diff
@@ -75,6 +75,12 @@ to using a real Chrome browser, which behaves exactly as a user's
 browser. Only use the `@javascript` tag when it's needed, to make the
 test suite run as fast as possible.
 
+Note that scenarios that involves file/image uploads is run with
```
Comment (naxoc) [inline features/README.md:78]: Grammar:
Note that scenarios that involves file/image uploads is run with
->
Note that scenarios that _involve_ file/image uploads _are_ run with

File: features/bootstrap/Page/Article.php [hunk 2]
```diff
@@ -13,4 +13,27 @@ public function hasTeaser($teaser): bool
...
+        // We look at the img inside the picture element.
+        $url = $this->findOrFail('css', '.node .field--name-image img')->getAttribute('src');
+
+        $urlInfo = parse_url($url);
+
+        if (!array_key_exists('path', $urlInfo)) {
+            throw new \RuntimeException('Could not parse main image URL: ' . $url);
+        }
+
+        if (!preg_match('{^/s3/files/styles/hero_desktop(?<imagepath>.*)$}', $urlInfo['path'], $matches)) {
```
Comment (naxoc) [inline features/bootstrap/Page/Article.php:28]: Would it make sense to leave out the `s3` part? To make the test care less about where things are stored?

File: features/bootstrap/Page/Common.php [hunk 3]
```diff
@@ -34,22 +43,30 @@ public function hasTitle(string $title): bool
...
+     * support.
      */
-    public function hasTeaser(string $teaser): bool
+    public function waitForAjaxToFinish()
     {
-        return $this->getTeaser() === $teaser;
+        $timeout = 5;
+        $condition = <<<JS
+    (function() {
+      function isAjaxing(instance) {
```
Comment (naxoc) [inline features/bootstrap/Page/Common.php:56]: LOL. ++ for ajaxing :rofl:

File: features/bootstrap/Page/Common.php [hunk 4]
```diff
@@ -34,22 +43,30 @@ public function hasTitle(string $title): bool
...
+        return instance && instance.ajaxing === true;
+      }
+      var d8_not_ajaxing = (typeof Drupal === 'undefined' || typeof Drupal.ajax === 'undefined' || typeof Drupal.ajax.instances === 'undefined' || !Drupal.ajax.instances.some(isAjaxing))
```
Comment (naxoc) [inline features/bootstrap/Page/Common.php:59]: Should be `const`.

File: features/bootstrap/Page/NodeAdd.php [hunk 5]
```diff
@@ -17,4 +19,45 @@ public function submit(): void
...
+        $preview = $this->findOrFail('css', '.image-preview img', $imageWidget);
+        $url = $preview->getAttribute('src');
+
+        $urlInfo = parse_url($url);
+
+        if (!array_key_exists('path', $urlInfo)) {
+            throw new \RuntimeException('Could not parse preview URL: ' . $url);
+        }
+
+        if (!preg_match('{^/s3/files/styles/4_3_100px(?<imagepath>.*)$}', $urlInfo['path'], $matches)) {
```
Comment (naxoc) [inline features/bootstrap/Page/NodeAdd.php:50]: Same with `s3`.

File: features/bootstrap/Page/Common.php (hunk 3 above)
Comment (xendk) [reply features/bootstrap/Page/Common.php:56]: Can't take credit for that, it's copied from drupal-extension.

**Pull request merged**